File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int num;
bool Comp(string a, string b){
    if(a[num] == b[num])
        return a < b;
    
    return a[num] < b[num];
}

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;

    num = n;
    sort(strings.begin(), strings.end(), Comp);
    
    answer = strings;
    
    return answer;
}

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
    var answer = [];
    
    strings.sort((a, b) => {
        if(a[n] === b[n])
            return a > b ? 1 : -1;
       
        return a[n] > b[n] ? 1 : -1;
        
    });
    
    answer = strings;
    
    return answer;
}